src/models/ComponentAttribute.js

```diff
@@ -84,6 +84,37 @@ class ComponentAttribute {
       || this.__ruleRegexValidation();
   }
 
+  /**
+   * Get the reference value of the attribute.
+   * @returns {(string | boolean | number | Array | ComponentAttribute[])}
+   * The value of the attribute.
+   */
+  getReferenceValue() {
+    return this.value;
+  }
+
+  /**
+   * Set the reference value of the attribute.
+   * @param {(string | boolean | number | Array | ComponentAttribute[])} value
+   * The new reference value to be set.
+   */
+  setReferenceValue(value) {
+    this.value = value;
+  }
+
+  /**
+   * Replace the old ID with the new ID in the link attribute's value.
+   * @param {string} oldId - The old ID.
+   * @param {string} newId - The new ID.
+   */
+  replaceLinkId(oldId, newId) {
+    const index = this.value.indexOf(oldId);
```

Comment:
If `this.value` is a boolean (cf. l.98), it should not work.

Reply:
a link can't be a boolean

Reply:
done